app/code/Magento/CatalogGraphQl/DataProvider/Product/LayeredNavigation/AttributeOptionProvider.php

```diff
@@ -90,7 +90,7 @@ public function getOptions(array $optionIds, ?int $storeId, array $attributeCode
                 Store::DEFAULT_STORE_ID
             );
 
-        $select->where('option_value.option_id IN (?)', $optionIds);
+        $select->where('a.attribute_code IN (?)', $attributeCodes);
```

Comment:
Using `$attributeCodes` instead of `$optionIds` makes the `$optionIds` unused. Considering the name of the method `getOptions` it seems highly unlikely that this should be the case.

In `Magento\CatalogGraphQl\DataProvider\Product\LayeredNavigation\Builder\Attribute` on line 158 this method is called with an array of option ids, these would be ignored.

Or maybe I'm wrong (I haven't looked to much into the issue) and this is the proper solution. In that case I'd like to ask to consider refactoring the method to remove `$optionIds` as an argument to prevent confusion in the future (might someone look at the method and ask the question "is this unused variable simply unused or should it be used and thus a bug").

Reply:
@Serializator could you please suggest the required solution for the same?

Reply:
@ihor-sviziev could you please suggest the required solution for issue #28878 ?

Reply:
@Serializator could you please review my last commit, I have made required changes.

Reply:
@rani-webkul, good pull request 👍 🎉

Reply:
@Serializator Thankyou for approving the changes.
But as I can see this pull request is not passing all the checks. Could you please let me know, If I am missing something?

Reply:
@rani-webkul, I only see that the tests are waiting for a status report, none are failing yet.

Reply:
@Serializator it's been 6 days but checks are not completing why?